tests/rptest/tests/throughput_limits_snc_test.py

```diff
@@ -0,0 +1,180 @@
+
+        if prop == self.ConfigProp.QUOTA_SHARD_MIN_RATIO:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            if r == 1:
+                return 1
+            return self.rnd.random()
+
+        if prop == self.ConfigProp.THROTTLE_DELAY_MAX_MS:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            return math.floor(2**(self.rnd.random() * 25))  # up to ~1 year
+
+        if prop == self.ConfigProp.BAL_WINDOW_MS:
+            r = self.rnd.randrange(4)
+            if r == 0:
+                return 1
+            if r == 1:
+                return 2147483647
+            return math.floor(2**(self.rnd.random() * 31))
+
+        if prop == self.ConfigProp.BAL_PERIOD_MS:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            return math.floor(2**(self.rnd.random() * 22))  # up to ~1.5 months
+
+        raise Exception(f"Unsupported ConfigProp: {prop}")
+
+    def setUp(self):
+
+        self.config = {}
+        for prop in list(self.ConfigProp):
+            val = self.get_config_parameter_random_value(prop)
+            self.config[prop] = val
+            if not val is None:
+                self.redpanda.add_extra_rp_conf({prop.value: val})
+
+        self.logger.info(
+            f"!SETTINGUP RP! extras:{self.redpanda._extra_rp_conf}")
+        super(ThroughputLimitsSnc, self).setUp()
+
+    def current_effective_node_quota(self) -> Tuple[int, int]:
+        metrics = self.redpanda.metrics_sample(
+            "quotas_snc_quota_effective",
+            metrics_endpoint=MetricsEndpoint.METRICS)
+
+        assert metrics, "Effecive quota metric is missing"
+        self.logger.debug(f"Samples: {metrics.samples}")
+
+        node_quota_in = sum(
+            int(s.value) for s in metrics.label_filter({
+                "direction": "ingress"
+            }).samples)
+        node_quota_eg = sum(
+            int(s.value) for s in metrics.label_filter({
+                "direction": "egress"
+            }).samples)
+        return node_quota_in, node_quota_eg
+
+    @cluster(num_nodes=3)
+    def test_configuration(self):
+        """
+        Test various configuration patterns, including extreme ones,
+        verify that it does not wreck havoc onto cluster
+        """
+
+        # TBD: parameterize to run under load or not
+
+        errors = 0
+        run_until = time.clock_gettime(time.CLOCK_MONOTONIC) + 15
+        # seconds
+        while time.clock_gettime(time.CLOCK_MONOTONIC) < run_until:
+
+            config_param = self.rnd.choice(list(self.ConfigProp))
+            config_value = self.get_config_parameter_random_value(config_param)
+            self.logger.debug(f"Setting {config_param.value} = {config_value}")
+            self.config[config_param] = config_value
+            self.rpk.cluster_config_set(config_param.value,
```

Comment:
Are all we doing here is setting config values and observing the effect? I would expect some partitions to be created and traffic to be sent so we can observe how the quotas move around from shard to shard

Reply:
Yes so far we're setting parameters at random, checking that it doesn't blow the cluster up, and also some obvious metrics. Testing for the actual results of the balancing will be the next PR with tests.